### DISCUSSION TITLE
If I look at the latest release, which is marked as latest, but is not the last published one, does the source code work for all templates?

### DISCUSSION THREAD

**Discussion by devopsaz-owner** [actions/virtual-environments#9198] [General questions] [answered]
I build Win22 and Ubuntu22 images for our ADO pipes.  I have been using the main branch to source our images from, which may present issues as some items are beta.  I see releases has 'Ubuntu 22.04 (20240107) Image Update' listed as latest, published 1/7, but I see 'Windows Server 2022 (20240116) Image Update' published 1/17 and 'Ubuntu 22.04 (20240116) Image Update' published 1/16.  

My assumption is I don't have to grab the windows release for the windows build and the ubuntu release for the ubuntu build, but can use the "Latest" release for both.  

Please correct me if I am wrong.

**Comment by shamil-mubarakshin** [comment, marked as answer]
Hey @DevOpsAzurance,
We generally avoid installing Beta and RC software.
While certain tools have their versions pinned due to various reasons, software installation often uses latest available non-prerelease version. This and dependencies introduced by package managers (apt, brew, choco) often causes image builds run from tags/releases to diverge from tag/release info or even fail. 
Thus we always encourage usage of  `main` branch as a source code for image builds. Our team performs continuous work on fixing bugs, updating software and ensuring image builds run successfully.
